File: android/core/src/main/java/com/novoda/bonfire/user/service/PersistedUserService.java
package com.novoda.bonfire.user.service;

import com.novoda.bonfire.user.data.model.User;
import com.novoda.bonfire.user.data.model.Users;
import com.novoda.bonfire.user.database.UserDatabase;

import rx.Observable;

public class PersistedUserService implements UserService {

    private final UserDatabase userDatabase;

    public PersistedUserService(UserDatabase userDatabase) {
        this.userDatabase = userDatabase;
    }

    @Override
    public Observable<Users> getAllUsers() {
        return userDatabase.observeUsers();
    }

    @Override
    public Observable<User> getUser(String userId) {
        return userDatabase.observeUser(userId);
    }

}

File: ios/Bonfire/Core/Users/PersistedUserService.swift
import RxSwift

class PersistedUserService: UsersService {

    private let userDatabase: UserDatabase

    init(userDatabase: UserDatabase) {
        self.userDatabase = userDatabase
    }

    func allUsers() -> Observable<Users> {
        return userDatabase.observeUsers()
    }

    func user(userID: String) -> Observable<User> {
        return userDatabase.observeUser(userID)
    }
}
